deploy/pivot_role/pivotRoleCDK/dataall_base_infra.py

```diff
@@ -0,0 +1,827 @@
+                                         actions=[
+                                             "iam:PutRolePolicy",
+                                             "iam:DeleteRolePolicy"
+                                         ],
+                                         resources=["*"]
+                                     ),
+                                     iam.PolicyStatement(
+                                         sid="IAMPassRole", effect=iam.Effect.ALLOW,
+                                         actions=[
+                                             "iam:Get*",
+                                             "iam:PassRole"
+                                         ],
+                                         resources=[
+                                             f"arn:aws:iam::{self.account}:role/{env_resource_prefix}*",
+                                             f"arn:aws:iam::{self.account}:role/{role_name}",
+                                             f"arn:aws:iam::{self.account}:role/cdk-*"
+                                         ],
+                                     ),
+                                     # STS
+                                     iam.PolicyStatement(
+                                         sid="STS", effect=iam.Effect.ALLOW,
+                                         actions=["sts:AssumeRole"],
+                                         resources=[f"arn:aws:iam::{self.account}:role/{env_resource_prefix}*"]
+                                     ),
+                                     # Step Functions
+                                     iam.PolicyStatement(
+                                         sid="StepFunctions", effect=iam.Effect.ALLOW,
+                                         actions=[
+                                             "states:DescribeStateMachine",
+                                             "states:ListExecutions",
+                                             "states:StartExecution"
+                                         ],
+                                         resources=[
+                                             f"arn:aws:states:*:{self.account}:stateMachine:{env_resource_prefix}*"]
+                                     ),
+                                     # CodeCommit
+                                     iam.PolicyStatement(
+                                         sid="CodeCommit", effect=iam.Effect.ALLOW,
+                                         actions=[
+                                             "codecommit:GetFile",
+                                             "codecommit:ListBranches",
+                                             "codecommit:GetFolder",
+                                             "codecommit:GetCommit",
+                                             "codecommit:GitPull",
+                                             "codecommit:GetRepository"
+                                         ],
```

Comment:
Need the following codecommit permissions for pipelines to deploy sucessfully
              - 'codecommit:TagResource'
              - 'codecommit:UntagResource'
              - 'codecommit:CreateBranch'
              - 'codecommit:CreateCommit'
              - 'codecommit:CreateRepository'
              - 'codecommit:DeleteRepository'
              - 'codecommit:GitPush'
              - 'codecommit:PutFile'
              - 'codecommit:GetBranch'